### PR TITLE
add error handler for  podContainerChanges.ContainersToStart in kubelet

### LINKED ISSUE #104496: Add error handler for  podContainerChanges.ContainersToStart in kubelet
#### What would you like to be added:
https://github.com/kubernetes/kubernetes/blob/3a26b864f4e17a6b7e0e673f583a551820a25b36/pkg/kubelet/kuberuntime/kuberuntime_manager.go#L929
add error handler for  podContainerChanges.ContainersToStart in kubelet
#### Why is this needed:
Errors should be actively handled in golang. Thanks very much

### PR DESCRIPTION
**Pull request by yxxhero**
Signed-off-by: yxxhero <aiopsclub@163.com>

#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
add error handler for  podContainerChanges.ContainersToStart in kubelet.

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/kubernetes/issues/104496

#### Special notes for your reviewer:
NONE

#### Does this PR introduce a user-facing change?
NONE

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/kubelet/kuberuntime/kuberuntime_manager.go [hunk 1]
```diff
@@ -926,7 +926,10 @@ func (m *kubeGenericRuntimeManager) SyncPod(pod *v1.Pod, podStatus *kubecontaine
 
 	// Step 7: start containers in podContainerChanges.ContainersToStart.
 	for _, idx := range podContainerChanges.ContainersToStart {
-		start("container", metrics.Container, containerStartSpec(&pod.Spec.Containers[idx]))
+		if err := start("container", metrics.Container, containerStartSpec(&pod.Spec.Containers[idx])); err != nil {
+			return
+		}
+		klog.V(4).InfoS("Completed container for pod", "containerName", &pod.Spec.Containers[idx].Name, "pod", klog.KObj(pod))
```
Comment (cndoit18) [inline pkg/kubelet/kuberuntime/kuberuntime_manager.go:932]:
```suggestion
		err = start("container", metrics.Container, containerStartSpec(&pod.Spec.Containers[idx]))
		utilruntime.HandleError(err)
```

Comment (cndoit18) [reply pkg/kubelet/kuberuntime/kuberuntime_manager.go:932]: the return may lead to a change in logic.

**Review by yangjunmyfm192085 — COMMENTED** [review]
I think the reason why there is no print  error is to prevent a lot of logs.
If want to modify, you can do the following test
- When the container restarts repeatedly,if it can generate a lot of logs
- EphemeralContainers Can be optimized both.

/cc @ehashman @249043822

File: pkg/kubelet/kuberuntime/kuberuntime_manager.go [hunk 2]
```diff
@@ -926,7 +926,8 @@ func (m *kubeGenericRuntimeManager) SyncPod(pod *v1.Pod, podStatus *kubecontaine
 
 	// Step 7: start containers in podContainerChanges.ContainersToStart.
 	for _, idx := range podContainerChanges.ContainersToStart {
-		start("container", metrics.Container, containerStartSpec(&pod.Spec.Containers[idx]))
+		err = start("container", metrics.Container, containerStartSpec(&pod.Spec.Containers[idx]))
+		utilruntime.HandleError(err)
```
Comment (249043822) [inline pkg/kubelet/kuberuntime/kuberuntime_manager.go:930]: nit: this change seems duplicate with https://github.com/kubernetes/kubernetes/blob/3a26b864f4e17a6b7e0e673f583a551820a25b36/pkg/kubelet/kuberuntime/kuberuntime_manager.go#L898 inside

Comment (yxxhero) [reply pkg/kubelet/kuberuntime/kuberuntime_manager.go:930]: yes, you are right. I will think it again. Thanks very much.

**Pull request closed without merging**